### PR TITLE
Fixed EfficientNetV2's MBConvBlock output phase.

### LINKED ISSUE #17358: Error in EfficientNet V2 MBConvBlock
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): No.
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.11
- Python version: 3.9
- Bazel version (if compiling from source): N/A
- GPU model and memory: N/A
- Exact command to reproduce: N/A

**Describe the problem**.
There is an error in EfficientNetV2's MBConvBlock - the final conv layers are not applied if Squeeze and Excitation block is not applied. This is an error, found by @DavidLandup0 in [Keras-CV repo](https://github.com/keras-team/keras-cv/pull/1146#discussion_r1050338581).

"Luckily" for MBConvBlock the SE Block is **always** applied so no need to change the weights.

**Describe the current behavior**.
Currently, the model works by accident.

**Describe the expected behavior**.
The MBConvBlock should work correctly even without SE Block.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**.

- Do you want to contribute a PR? (yes/no): yes I can.
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):
Basically, we need to unindent lines 717-740 in [efficientnet_v2.py](https://github.com/keras-team/keras/blob/master/keras/applications/efficientnet_v2.py#L717)

**Standalone code to reproduce the issue**.
N/A

**Source code / logs**.
Current faulty [MBConvBlock implementation](https://github.com/keras-team/keras/blob/9831c04766a227ae2b8787c982e2b63d9b1fb295/keras/applications/efficientnet_v2.py#L717).
Current [FusedMBConvBlock implementation](https://github.com/keras-team/keras/blob/9831c04766a227ae2b8787c982e2b63d9b1fb295/keras/applications/efficientnet_v2.py#L745), **without this issue**.
David's [fixed implementation](https://github.com/keras-team/keras-cv/blob/a3ae2dade8f536e96ee51f22b7e0a938cf2e14b1/keras_cv/layers/mbconv.py).

Thanks again @DavidLandup0 for finding this issue.

### PR DESCRIPTION
**Pull request by sebastian-sz**
Fixes #17358

Unindents output phase of MBConvBlock so that it is applied regardless of SE ratio.

To my knowledge, when inspecting the EfficientNetV2's configs, all MBConvBlocks use SE block part so the models and weights should be unchanged.

### CODE REVIEW

**Comment by fchollet** [comment]
Thanks for the PR. What validation accuracy do you see with the modified model on the imagenet test set?

CC @ianstenbit

**Comment by sebastian-sz** [comment]
@fchollet I did not run Imagenet evaluation as I suspect the models are unchanged.  For all variants, after fix:
1. The existing weights load without issue.
2. If running inference on the same image, these models output the same values as the ones in `tf-nightly`.

[gist](https://gist.github.com/sebastian-sz/b04d33eb7584efe3ada872e17686b1ee) I used for comparison.

**Comment by ianstenbit** [comment]
Thanks Sebastian and Francois -- looks like this fix will also need to be made in KerasCV's EfficientNetV2 (and it will be in https://github.com/keras-team/keras-cv/pull/1146)

**Comment by haifeng-jin** [comment]
Pending on another reply from @fchollet .

**Review by fchollet — APPROVED** [review]
LGTM, thanks!
